text/0000-wallet-replay-protection.md

```diff
@@ -0,0 +1,298 @@
+
+If we again used `O3` as an input in a new transaction `T3` to send money to Charlie, the same protection would be put in place. The attacker would need to replay `T2` to recreate `O3` input, but we have already shown that `T2` can't be replayed.
+
+```
+O3  -> O4 (change output)
+    -> Charlie_output
+```
+
+This is the main idea used in this proposal to protect against replay attacks. This is why we name the outputs we created along side the `anchor` output `Protected` outputs. They provide protection for subsequent transactions we make if we use them as inputs to our transactions. Key thing to note here is that while we started with 2 protected outputs `O1` and `O2`, the outputs that we create in a transaction to which we add a protected input automatically become `Protected` outputs as well. This property is what allows us to continue creating new protected outputs without the need to create more than one `anchor` output.
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Wallet rules for replay protection
+
+Most common transaction types are:
+1. A 1-2 `Regular` tx - 1 input, 2 outputs
+2. A 2-2 `PayJoin` tx - 2 inputs (one from each party), 2 outputs
+
+Let's call outputs that hang from a chain that leads to your anchor a `Protected` output. A transaction is protected from replays if a party contributes a Protected output as an input in the transaction. It's not enough that some other party contributes such an input. We are only really protected from replay attacks if we contribute the `Protected` output. Otherwise, our security depends on the honesty and following of the wallet rules of other participants which should not be relied on.
+
+Let's say that we want to be able to do both `Regular` and `PayJoin` transactions. We do both of these so we have some protected outputs and some that are not protected. The first problem we have is that after a wallet restore, we can't really tell whether an output is protected through a PayJoins+Anchor or not. It would be nice if we could label our outputs as either `Protected` or `Unprotected` after a wallet restore.
+
+### Separation of replay protected outputs
+
+Let's define a clear separation of outputs that are protected from those that are not. We define a protected output generator `GenP` that allows us to either `create` a protected output or `check` whether an output is `Protected`.
+
+An output is labeled as `Protected` *only if* it was created in a transaction where we contributed an input that was labeled as `Protected`. An exception to the rule are outputs that are created as a part of the _Bootstrap_ transaction discussed in the next section.
+
+#### Possible protected output generators implementations

```

Comment:
These are very helpful as outlined. Once you've decided to go in a specific direction, it would make sense to then move the other approaches to the "Rationale & Alternatives" section and explain why you opted not to use them.